tmt/steps/provision/__init__.py

```diff
@@ -200,6 +200,8 @@ def details(self) -> None:
         if self.opt('dry'):
             return
 
+        self.info('full name', self.full_name, 'green')
```

Comment:
Shouldn't this be called `name` to be consistent with what we are using the `provision` step config? I see we already are reporting another info under this key. For `container` provision the container name:

```
/plans/features/basic
summary: Basic command line features
    provision
        how: container
        image: fedora
        name: tmt-009-uwMrmtyU
        full name: default-0
        distro: Fedora Linux 37 (Container Image)
        kernel: 6.1.18-200.fc37.x86_64
        summary: 1 guest provisioned
```

For `virtual` provision the test cloud instance name:

```
/plans/default
    provision
        how: virtual
        user: root
        image: fedora
        memory: 2048 MB
        disk: 10 GB
        connection: session
        arch: x86_64
        qcow: Fedora-Cloud-Base-37-1.7.x86_64.qcow2
        name: tmt-010-qTznmpRY
        key: /var/tmp/tmt/run-010/plans/default/provision/default-0/id_ecdsa
        progress: booting...
        ip: 127.0.0.1
        port: 10023
        full name: default-0
```

I'd probably suggest to rename the other ones (e.g. use `container` for podman and `instance` for testcloud) to better match reality and keep consistency with the step phase names. What do you think?

Reply:
I'm not sure. IIUIC, `name` in the `provision` phase is the name of the phase, not necessarily the name of the guest because the "name of the guest" is something different plugins create in a different way.

I wanted something I could use to identify the guest in multi-host logs, hence "full name" - maybe we should rename this property to `multihost_name` since that was the original purpose, and it seems now it'd be a better fit for the content it provides.

I agree "name" should be changed a bit, e.g. `name` => `container name`, as that's exactly what it is, and `instance name` for `testcloud`. `artemis` plugins stays away from "name" and prints out `guestname`, which is good. Although I'd resolve the question of "name" in a different patch.

Reply:
> I'm not sure. IIUIC, `name` in the `provision` phase is the name of the phase...

Yes, and that should/could be consistent across all steps. But yes, here the full name is a different story. Let's fix the naming in a separate patch.